src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,250 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.CodeRefactorings.InvertIf
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.Operations
+Imports Microsoft.CodeAnalysis.Simplification
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InvertIf
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeRefactoringProviderNames.InvertIf), [Shared]>
+    Friend Class VisualBasicInvertIfCodeRefactoringProvider
+        Inherits AbstractInvertIfCodeRefactoringProvider
+
+        Private Shared ReadOnly s_ifNodeAnnotation As New SyntaxAnnotation
+
+        Protected Overrides Function GetSyntaxFactsService() As ISyntaxFactsService
+            Return VisualBasicSyntaxFactsService.Instance
+        End Function
+
+        Protected Overrides Function GetIfStatement(textSpan As TextSpan, token As SyntaxToken, cancellationToken As CancellationToken) As SyntaxNode
+
+            Dim relevantIfBlockOrIfStatement As ExecutableStatementSyntax = Nothing
+            Dim relevantSpan As TextSpan = Nothing
+
+            Dim singleLineIf = token.GetAncestor(Of SingleLineIfStatementSyntax)()
+            Dim multiLineIf = token.GetAncestor(Of MultiLineIfBlockSyntax)()
+
+            If singleLineIf IsNot Nothing AndAlso
+               singleLineIf.ElseClause IsNot Nothing Then
+
+                relevantIfBlockOrIfStatement = singleLineIf
+                relevantSpan = TextSpan.FromBounds(singleLineIf.IfKeyword.Span.Start, singleLineIf.Condition.Span.End)
+
+            ElseIf multiLineIf IsNot Nothing AndAlso
+                   multiLineIf.ElseBlock IsNot Nothing AndAlso
+                   multiLineIf.ElseIfBlocks.IsEmpty Then
+
+                relevantIfBlockOrIfStatement = multiLineIf
+                relevantSpan = TextSpan.FromBounds(multiLineIf.IfStatement.IfKeyword.Span.Start, multiLineIf.IfStatement.Condition.Span.End)
+            Else
+                Return Nothing
+            End If
+
+            If Not relevantSpan.IntersectsWith(textSpan.Start) Then
+                Return Nothing
+            End If
+
+            If token.SyntaxTree.OverlapsHiddenPosition(relevantSpan, cancellationToken) Then
+                Return Nothing
+            End If
```

Comment:
why do you need this check?  i still feel ike it's unnecessary given the similar check in the base class.